File: csharp/main/Implementation/InterfaceMethodNode.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;

namespace Deltix.Luminary.Implementation
{
	internal class InterfaceMethodNode : ItemNode
	{
		public InterfaceMethodNode([NotNull] String name, [CanBeNull] String returnType, [CanBeNull] List<String> comments)
			: base(name, comments)
		{
			ReturnType = returnType;
			FormalParameters = new List<FormalParameterNode>();
		}

		[CanBeNull]
		public String ReturnType { get; }

		public List<FormalParameterNode> FormalParameters { get; }
	}
}

File: java/src/main/java/deltix/luminary/implementation/InterfaceMethodNode.java
package deltix.luminary.implementation;

import java.util.ArrayList;
import java.util.List;

public class InterfaceMethodNode extends ItemNode {
    private final String returnType;
    private final List<FormalParameterNode> formalParameters;

    protected InterfaceMethodNode(String name, String returnType, List<String> comments) {
        super(name, comments);
        this.returnType = returnType;
        this.formalParameters = new ArrayList<>();
    }

    public String getReturnType() {
        return returnType;
    }

    public List<FormalParameterNode> getFormalParameters() {
        return formalParameters;
    }
}
